pkg/controller/machine/machine_controller.go

```diff
@@ -721,6 +724,27 @@ func (r *Reconciler) ensureInstanceExistsForMachine(
 				externalCloudProvider, _ = strconv.ParseBool(val)
 			}
 
+			registryCredentials := map[string]containerruntime.AuthConfig{}
+
+			if secRef := strings.SplitN(r.nodeSettings.RegistryCredentialsSecretRef, "/", 2); len(secRef) == 2 {
+				var credsSecret corev1.Secret
+				err := r.client.Get(ctx, types.NamespacedName{Namespace: secRef[0], Name: secRef[1]}, &credsSecret)
```

Comment:
This will require changes to RBAC so machine-controller can access the Secret. We need to somehow document that.